pkg/connector/roles.go

```diff
@@ -0,0 +1,225 @@
+	{ID: "organization_external_user", DisplayName: "Organization External User", RoleKey: "ORGANIZATION_EXTERNAL_USER"},
+	{ID: "organization_team_guest_user", DisplayName: "Team Guest User", RoleKey: "ORGANIZATION_TEAM_GUEST_USER"},
+}
+
+// roleBuilder is the builder for the role resource type.
+type roleBuilder struct {
+	client         *miro.Client
+	resourceType   *v2.ResourceType
+	organizationId string
+}
+
+// ResourceType returns the resource type for the role builder.
+func (r *roleBuilder) ResourceType(ctx context.Context) *v2.ResourceType {
+	return r.resourceType
+}
+
+// List returns the resources for the role builder.
+func (r *roleBuilder) List(ctx context.Context, parentResourceID *v2.ResourceId, pToken *pagination.Token) ([]*v2.Resource, string, annotations.Annotations, error) {
+	var resources []*v2.Resource
+	for _, role := range roleDefinitions {
+		profile := map[string]interface{}{
+			"role_id":   role.ID,
+			"role_name": role.DisplayName,
+		}
+		roleResource, err := resource.NewRoleResource(
+			role.DisplayName,
+			r.resourceType,
+			role.ID,
+			[]resource.RoleTraitOption{resource.WithRoleProfile(profile)},
+		)
+		if err != nil {
+			return nil, "", nil, fmt.Errorf("failed to create role resource: %w", err)
+		}
+		resources = append(resources, roleResource)
+	}
+	return resources, "", nil, nil
+}
+
+// Entitlements returns the entitlements for the role builder.
+func (r *roleBuilder) Entitlements(ctx context.Context, resource *v2.Resource, _ *pagination.Token) ([]*v2.Entitlement, string, annotations.Annotations, error) {
+	assigmentOptions := []entitlement.EntitlementOption{
+		entitlement.WithGrantableTo(userResourceType),
+		entitlement.WithDescription(fmt.Sprintf("%s to %s role", assignedRole, resource.DisplayName)),
+		entitlement.WithDisplayName(fmt.Sprintf("%s role %s", resource.DisplayName, assignedRole)),
+	}
+
+	ent := entitlement.NewAssignmentEntitlement(
+		resource,
+		assignedRole,
+		assigmentOptions...,
+	)
+
+	return []*v2.Entitlement{ent}, "", nil, nil
+}
+
+// Grants returns the grants for the role builder - users who have this specific role.
+func (r *roleBuilder) Grants(ctx context.Context, resource *v2.Resource, pToken *pagination.Token) ([]*v2.Grant, string, annotations.Annotations, error) {
+	roleID := resource.Id.Resource
+
+	var roleKey string
+	for _, definition := range roleDefinitions {
+		if definition.ID == roleID {
+			roleKey = definition.RoleKey
+			break
+		}
+	}
+
+	if roleKey == "" {
+		return nil, "", nil, fmt.Errorf("role key not found for role ID: %s", roleID)
+	}
+
+	var allUsers []miro.User
+	cursor := ""
+
+	for {
+		response, _, err := r.client.GetOrganizationMembers(ctx, r.organizationId, cursor, resourcePageSize)
+		if err != nil {
+			return nil, "", nil, fmt.Errorf("failed to get organization members: %w", err)
+		}
+
+		allUsers = append(allUsers, response.Data...)
+
+		if response.Cursor == "" {
+			break
+		}
+		cursor = response.Cursor
```

Comment:
this is making the same API call per role, over and over again. It is also paginating in a loop without returning ever returning the pagination token. How about we emit this grant from the user resource instead